common/ts/__tests__/publish-task-test.ts

```diff
@@ -60,8 +60,18 @@ it('check multiple report status and set global quality gate for build propertie
   taskReportArray.push(taskReport);
   taskReportArray.push(taskReport);
 
-  jest.spyOn(Task, 'waitForTaskCompletion').mockImplementation(() => returnedTaskOk);
-  jest.spyOn(TaskReport, 'createTaskReportsFromFiles').mockImplementation(() => taskReportArray);
+  jest.spyOn(Task, 'waitForTaskCompletion').mockImplementation(
+    () =>
+      new Promise<Task>(resolve => {
+        return resolve(returnedTaskOk);
```

Comment:
You can do things like `jest.spyOn(Task, 'waitForTaskCompletion').mockResolvedValue(returnedTaskOk);` and it should have the same effect.

Reply:
Thanks, that works !